### PR TITLE
gh-274: add missing tests for `glass.user`

### LINKED ISSUE #274: Add/review tests for `glass.user`
### Is Your Feature Request Related to a Problem? Please Describe

Part of #161

### Describe the Solution You'd Like

_No response_

### Describe Alternatives You've Considered

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Saransh-cpp**
Closes #274. Adds tests for `glass.save_cls` and `glass.load_cls`.

### CODE REVIEW

File: tests/test_fits.py
Comment (Saransh-cpp) [inline tests/test_fits.py:1]: The file is named `tests/test_fits.py` but all the `fits` related functions are present in `glass.user`. Should this be renamed?

File: tests/test_fits.py [hunk 1]
```diff
@@ -14,6 +14,25 @@
...
+    with np.load(tmp_path / cls_file) as npz:
+        values = npz["values"]
+        split = npz["split"]
+
+    np.testing.assert_allclose(values, np.concatenate(cls))
+    np.testing.assert_allclose(split, np.cumsum([len(cl) for cl in cls[:-1]]))
+    np.testing.assert_allclose(cls, np.split(values, split))
+
+    npz = load_cls(tmp_path / cls_file)
+    np.testing.assert_allclose(npz, cls)
```
Comment (Saransh-cpp) [inline tests/test_fits.py:35]: I could not find a better way of testing these functions in 2 independent tests. I could do that if I don't use the `tmp_path` fixture. Both cases have pros and cons, but I don't have any strong opinions.

Comment (ntessore) [reply tests/test_fits.py:35]: I think this is fine, we aren't testing that we adhere to an external file format here, so we don't need to test that `numpy.savez()` doesn't change.

File: tests/test_fits.py
Comment (ntessore) [reply tests/test_fits.py:1]: Makes sense!

File: tests/test_fits.py [hunk 2]
```diff
@@ -14,6 +14,25 @@
...
+    cls = rng.normal(size=(10, 10))
+    save_cls(tmp_path / cls_file, cls)
+
+    assert pathlib.Path.exists(tmp_path / cls_file)
+
+    with np.load(tmp_path / cls_file) as npz:
+        values = npz["values"]
+        split = npz["split"]
+
+    np.testing.assert_allclose(values, np.concatenate(cls))
```
Comment (paddyroddy) [inline tests/test_fits.py:30]: Why `assert_allclose` here? Wouldn't https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html be more appropriate?

Comment (Saransh-cpp) [reply tests/test_fits.py:30]: Oh yes, thanks!

**Review by paddyroddy — APPROVED** [review]
Nice!

**Pull request merged**